app/jobs/generate_job.rb

```diff
@@ -3,6 +3,12 @@ class GenerateJob < ActiveJob::Base
 
   queue_as MarkusConfigurator.markus_job_generate_queue_name
 
+  def self.on_complete_js(status)
+    path = Rails.application.routes.url_helpers.download_generate_assignment_exam_template_path(assignment_id: status[:id],
+                                                                                                id: status[:exam_id])
+    "window.location=\"#{path}?file_name=#{status[:file_name]}\""
```

Comment:
Also, please put the `function () {...}` here - I'd rather see the entire function callback in this method than see it partly here, and partly in `poll_job`. This changes what you did for the other file as well.